src/components/TopBar.jsx

```diff
@@ -0,0 +1,175 @@
+/* global _ */
+import React, { useEffect, useState, useRef } from 'react';
+import cx from 'classnames';
+import Telemetry from 'src/libs/telemetry';
+import Suggest from 'src/components/ui/Suggest';
+import Menu from 'src/panel/Menu';
+import { useConfig, useDevice } from 'src/hooks';
+import { handleFocus } from 'src/libs/input';
+import { selectItem, fetchSuggests } from 'src/libs/suggest';
+
+const MAPBOX_RESERVED_KEYS = ['ArrowLeft', 'ArrowUp', 'ArrowRight', 'ArrowDown', '-', '+', '='];
+
+const TopBar = ({ value, setUserInputValue, inputRef, onSuggestToggle, backButtonAction }) => {
+  const suggestElement = useRef(null);
+  const [focused, setFocused] = useState(false);
+  const { isMobile } = useDevice();
+  const config = useConfig();
+
+  // give keyboard focus to the field when typing anywhere
+  useEffect(() => {
+    const globalKeyHandler = e => {
+      if (MAPBOX_RESERVED_KEYS.find(key => key === e.key)) {
+        return;
+      }
+      // KeyboardEvent.key is either the printed character representation or a standard value for specials keys
+      // See https://developer.mozilla.org/fr/docs/Web/API/KeyboardEvent/key/Key_Values
+      if (
+        e.key.length === 1 &&
+        !e.ctrlKey &&
+        !e.metaKey &&
+        document.activeElement?.tagName !== 'INPUT'
+      ) {
+        setUserInputValue(inputRef.current.value + e.key);
+        inputRef.current.focus();
```

Comment:
It works but I am not sure to understand why calling `setUserInputValue()` is necessary.  
Isn't the input value supposed to change (caught by `onChange`) after the input has been focused ?

Reply:
This is the point which is driving me mad…
The focus is done and yes this is supposed to work like you say, but the first `onChange` isn't triggered and the first letter is "lost". Honestly I've not figured exactly why yet, after a long debugging time spent on it…
I think it's related to the management of the user input value in `PanelManager`, as in one of the previous commits it was managed as local state in `TopBar` and it worked fine.

Feel free to spend time on it, maybe you'll understand what I wasn't able to, and I would be very interested in another PoV.

But if this is the last remaining hack I guess it's not too bad compared to the previous implementation.

Reply:
I am still not sure about what is going on here. Has this behavior ever worked with a "controlled" input? It looks like 
React manages a custom lifecycle for keystrokes in this case.

An alternative solution would be to re-dispatch the "keydown" event after the input has been focused. Then, the dependency to `setUserInputValue` could be removed:
```suggestion
        inputRef.current.focus();
        inputRef.current.dispatchEvent(new KeyboardEvent(e.type, e));
```
It may be more consistent with the initial implementation, but I am not sure it is any better.